src/UseCases/RegularVendor/AddShopUseCase.java

```diff
@@ -21,11 +21,11 @@ public AddShopUseCase(VendorRepository vendorRepository){
      *
      * @param shopId The id of the shop that we want to add to the vendor's shoplist.
      * @param token The token of the vendor.
-     * @return A boolean of whether the method successfully added the shop.
+     * @return whether the method successfully added the shop.
      */
     @Override
-    public boolean addShop(String shopId, String token) {
-        RegularVendor curr_ven = vendorRepository.getUserFromToken(token);
+    public boolean addShop(IShop shop, String token) {
+        IVendor curr_ven = vendorRepository.getVendorFromToken(token);
```

Comment:
This will cause an error - I believe it should be casted as
```suggestion
        IVendor curr_ven = (IVendor) vendorRepository.getUserFromToken(token);
```